### PR TITLE
Fixes/invalid yield

### LINKED ISSUE #319: Ruby 3.0 yield in class syntax warning
Running metric_fu version 4.13.0 from ruby version >= 2.7 we get:

> /usr/local/lib/ruby/gems/2.7.0/gems/metric_fu-4.13.0/lib/metric_fu/loader.rb:37: warning: `yield' in class syntax will not be supported from Ruby 3.0. [Feature #15575]
/usr/local/lib/ruby/gems/2.7.0/gems/metric_fu-4.13.0/lib/metric_fu/loader.rb:54: warning: `yield' in class syntax will not be supported from Ruby 3.0. [Feature #15575]

The following patch will fix these warnings in loader.rb:

```
--- /usr/local/lib/ruby/gems/2.7.0/gems/metric_fu-4.13.0/lib/metric_fu/loader.rb	2020-06-20 19:19:32.000000000 -0700
+++ /tmp/loader.rb	2020-06-20 19:19:17.000000000 -0700
@@ -33,14 +33,12 @@
     #     ::metrics_dir which returns the full path
     #     ::metrics_require which takes a block of files to require relative to the metrics_dir
     def create_dirs(klass)
-      class << klass
         Array(yield).each do |dir|
-          define_method("#{dir}_dir") do
-            File.join(lib_dir, dir)
+          klass.define_singleton_method("#{dir}_dir") do
+            File.join(klass.lib_dir, dir)
           end
-          module_eval(%(def #{dir}_require(&block); lib_require('#{dir}', &block); end),  __FILE__, __LINE__)
+          klass.module_eval(%(def self.#{dir}_require(&block); lib_require('#{dir}', &block); end),  __FILE__, __LINE__)
         end
-      end
     end
 
     # Adds method x_dir relative to the metric_fu artifact directory to the given klass
@@ -50,11 +48,9 @@
     # @example For the artifact sub-directory '_scratch', creates on the klass one method:
     #     ::scratch_dir (which notably has the leading underscore removed)
     def create_artifact_subdirs(klass)
-      class << klass
-        Array(yield).each do |dir|
-          define_method("#{dir.gsub(/[^A-Za-z0-9]/, '')}_dir") do
-            File.join(artifact_dir, dir)
-          end
+      Array(yield).each do |dir|
+        klass.define_singleton_method("#{dir.gsub(/[^A-Za-z0-9]/, '')}_dir") do
+          File.join(artifact_dir, dir)
         end
       end
     end
```
These two warnings are telling us that two methods are yielding after changing into the scope of another class:
```
class Foo
  def foo(klass)
    class << klass
      yield
...
```
My fix does does the yield without moving the body of the method into the scope of 'klass'.

### PR DESCRIPTION
**Pull request by etagwerker**
Hi there, 

This PR fixes #319 because the current syntax does not work with Ruby 3.0

Please check it out.

Thanks!

### CODE REVIEW

File: lib/metric_fu/loader.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def lib_require(base = "", &_block)
     #     ::metrics_require which takes a block of files to require relative to the metrics_dir
     def create_dirs(klass)
       class << klass
-        Array(yield).each do |dir|
```
Comment (bf4) [inline lib/metric_fu/loader.rb:37]: did it just need a `create_dirs(klass, &block)` instead maybe?

Comment (etagwerker) [reply lib/metric_fu/loader.rb:37]: yeah, I guess that could have worked too. I thought it was a good opportunity to simplify unnecessary indirection...

**Comment by pedrofurtado** [comment]
@etagwerker @lubc Is this change released in rubygems? 🤝

**Comment by lubc** [comment]
@pedrofurtado hey Pedro! I don't think so. I'll talk to my team tomorrow to see if we can release a new version 👍

**Comment by pedrofurtado** [comment]
@lubc Thanks a lot!

File: lib/metric_fu/loader.rb (hunk 1 above)
Comment (daveinman) [reply lib/metric_fu/loader.rb:37]: I'm not sure anyone cares, but this fix changes semantics.  One could, with the old code, override the library_dirs in use without a warning of overriding a constant (likewise for artifact_subdirs, I think).  To this fixes credit, it is simpler.  I don't know if any of these directories are subject to the whim of the tools metric_fu calls.  If they are,  then keeping the old semantics seems better.

**Pull request merged**